src/plugins/dna_export/src/ExportSequenceTask.cpp

```diff
@@ -553,18 +553,65 @@ U2Sequence ExportAnnotationSequenceSubTask::importAnnotatedSeq2Dbi(const SharedA
     importer.startSequence(os, resultDbiRef, U2ObjectDbi::ROOT_FOLDER, ad->name, false);
     CHECK_OP(os, U2Sequence());
 
-    foreach (const U2Region &annotatedRegion, ad->location->regions) {
+    for (const U2Region &annotatedRegion : qAsConst(ad->location->regions)) {
+        CHECK_CONTINUE(!annotatedRegion.isEmpty());
+
         qint64 currentRegionLength = 0;
-        for (qint64 pos = annotatedRegion.startPos; pos < annotatedRegion.endPos(); pos += MAX_CHUNK_LENGTH) {
-            const qint64 currentChunkSize = qMin(MAX_CHUNK_LENGTH, annotatedRegion.endPos() - pos);
+
+        // Go in the direct order in case of the direct sequence
+        // and in the reverse order in case of the reverse-complementary sequence
+        bool isReverseComplement = ad->getStrand().isCompementary() && ei.complTT != nullptr;
+        int pos = 0;
+        auto annotationEndPos = annotatedRegion.endPos();
+        // If we translate the reverse-complement sequence, we need to go from the rnd to the beginning
+        // And we need to skip first 1 or 2 bases (or don't skip them if the annotation length is divided by 3
+        int offset = 0;
+        qint64 currentChunkSize = qMin(MAX_CHUNK_LENGTH, annotationEndPos - pos);
+        if (isReverseComplement) {
+            offset = annotatedRegion.length % 3;
+            pos = annotationEndPos - MAX_CHUNK_LENGTH;
+            if (pos < annotatedRegion.startPos + offset) {
+                pos = annotatedRegion.startPos + offset;
+            }
+        } else {
+            pos = annotatedRegion.startPos;
+        }
+        auto condition = [&]() -> bool {
```

Comment:
nit: You share external state with a big 'condition' & 'change' function that internally have a clear 
"if (isReverseComplement) else" switches.
I would suggest running 2 loops: direct and reverse instead to make the code easier to read.

Reply:
I have been thinking about the way to solve this problem and I like the lambda-solutions more, I don't see a point of replacing it